library/general/src/lib/yast2/log_dir_rotator.rb

```diff
@@ -0,0 +1,196 @@
+# encoding: utf-8
+
+# ***************************************************************************
+#
+# Copyright (c) 2018-2019 SUSE LLC
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com.
+#
+# ***************************************************************************
+
+require "yast"
+require "fileutils"
+
+Yast.import "Mode"
+Yast.import "Directory"
+
+module Yast2
+  # The class LogDirRotator handels creating, rotating and removing
+  # directories for logging in the YaST2 log directory according to
+  # this schema:
+  #
+  #   rm #{NAME}-03
+  #   mv #{NAME}-02 #{NAME}-03
+  #   mv #{NAME}-01 #{NAME}-02
+  #   mv #{NAME}    #{NAME}-01
+  #
+  # System error are logged and no exception is raised by the functions.
+  #
+  # TODO This class is based on dump_manager.rb and could be used there.
+  class LogDirRotator
+    include Yast::Logger
+
+    # The number of old log directories to keep. This is in addition
+    # to the current one. Child classes can override the value. Max
+    # value is 99.
+    KEEP_OLD_LOG_DIRS = 3
+
+    # The name of the directory in /var/log/YaST2 to create, rotate
+    # and delete. Child classes should override the value.
+    NAME = "undefined".freeze
+
+    # Prepare (create, rotate and delete) the log directory.
+    def prepare
+      return if Yast::Mode.test
```

Comment:
Why is this line needed? If you, for some reason, need to keep it, please document in the method's description.

Reply:
my guess is that ProductControl is used in old testsuite and you want to skip this file modification in old testsuite.

Reply:
Because otherwise the old testsuite fails. But I can also document that.

Reply:
Thanks, that's what I wanted to know. Mentioning it would be fine so I do not ask the question again in the future :)